logbook-core/src/main/java/org/zalando/logbook/BodyFilters.java

```diff
@@ -52,15 +61,32 @@ public static BodyFilter accessToken() {
      */
     @API(status = MAINTAINED)
     public static BodyFilter replaceJsonStringProperty(final Set<String> properties, final String replacement) {
-        final String regex = properties.stream()
-                .map(Pattern::quote)
-                .collect(joining("|"));
-
         final Predicate<String> json = MediaTypeQuery.compile("application/json", "application/*+json");
+
+        final String regex = properties.stream().map(Pattern::quote).collect(joining("|"));
         final Pattern pattern = Pattern.compile("(\"(?:" + regex + ")\"\\s*:\\s*)\".*?\"");
+        final UnaryOperator<String> delegate = body -> pattern.matcher(body).replaceAll("$1\"" + replacement + "\"");
+
+        return (contentType, body) -> json.test(contentType) ? delegate.apply(body) : body;
+    }
+
+    /**
+     * Creates a {@link BodyFilter} that replaces the properties in the form url encoded body with given replacement.
+     *
+     * @param properties  query names properties to replace
+     * @param replacement String to replace the properties values
+     * @return BodyFilter generated
+     */
+    @API(status = MAINTAINED)
+    public static BodyFilter replaceFormUrlEncodedQuery(final Set<String> properties, final String replacement) {
+        final Predicate<String> formUrlEncoded = MediaTypeQuery.compile("application/x-www-form-urlencoded");
+
+        final QueryFilter delegate = properties.stream()
+                .map(name -> QueryFilters.replaceQuery(name, replacement))
+                .reduce(QueryFilter::merge)
+                .orElseGet(QueryFilter::none);
 
-        return (contentType, body) -> json.test(contentType) ?
-                pattern.matcher(body).replaceAll("$1\"" + replacement + "\"") : body;
+        return (contentType, body) -> formUrlEncoded.test(contentType) ? delegate.filter(body) : body;
```

Comment:
Should we have a private, shared method that takes a Predicate and a UnaryOperator and produces a BodyFilter?

Reply:
I don't see some reasons to do that - ternary operator quite readable itself.